src/w3d/lib/hashtemplate.h

```diff
@@ -41,249 +42,194 @@ template<typename Key, typename Value> class HashTemplateClass
     };
 
 public:
-    HashTemplateClass() : m_index(nullptr), m_table(nullptr), m_nextFree(-1), m_size(0) {}
+    HashTemplateClass() : m_hash(nullptr), m_table(nullptr), m_first(-1), m_size(0) {}
 
     ~HashTemplateClass()
     {
-        delete[] m_index;
+        delete[] m_hash;
         delete[] m_table;
     }
 
-    Value *Get(const Key &key) const
+    Value Get(const Key &key) const
```

Comment:
Is this really a Get that performs a copy? I would have thought it would be a `const Value&`

Reply:
We have specific information that this absolutely returns a Value and not a Value& or a const Value& (for example in the original game MeshModelClass::Init_For_NPatch_Rendering uses a HashTemplateClass<TriangleSide,SideIndexInfo> and I have explicitly seen a call to get for this HashTemplateClass making a copy of the SideIndexInfo rather than moving around a pointer)